packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-oidc-fields.ts

```diff
@@ -2971,5 +2971,599 @@ export const testOIDCFieldAuth = (engine: ImportedRDSType): void => {
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      expect(onUpdateSubscriptionResult[0].data[`onUpdate${modelName}`].privateContent).toEqual(todoRandomUpdated.privateContent);
+      expect(onUpdateSubscriptionResult[0].data[`onUpdate${modelName}`].customGroup).toEqual(todoRandom.customGroup);
+      expect(onUpdateSubscriptionResult[0].data[`onUpdate${modelName}`].customGroups).toEqual(todoRandomUpdated.customGroups);
+      expectNullFields(onUpdateSubscriptionResult[0].data[`onUpdate${modelName}`], ['ownersContent', 'adminContent', 'groupsContent']);
+
+      const onDeleteSubscriptionResult = await subTodoHelper.subscribe('onDelete', [], {}, user1UpdateAllowedSet, false);
+      expect(onDeleteSubscriptionResult).toHaveLength(0);
+    });
+
+    test('Model with renamed protected fields and restricted operations', async () => {
+      const modelName = 'TodoRenamedFields';
+      const user1TodoHelper = user1ModelOperationHelpers[modelName];
+      const user2TodoHelper = user2ModelOperationHelpers[modelName];
+
+      const todoPrivateFields = {
+        author: userName1,
+        authors: [userName1],
+        customGroup: adminGroupName,
+        customGroups: [adminGroupName],
+        privateContent: 'Private Content',
+      };
+      const createResultSetName = `create${modelName}`;
+      const updateResultSetName = `update${modelName}`;
+      const privateResultSet = `
+        id
+        author
+        authors
+        customGroup
+        customGroups
+        privateContent
+      `;
+
+      // cannot create a record with dynamic owner list protected field that does not allow create operation
+      await expect(
+        async () => await user1TodoHelper.create(createResultSetName, { ...todoPrivateFields, ownersContent: 'Owners Content' }),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(createResultSetName, 'Mutation'));
+
+      // cannot create a record with dynamic groups list protected field that does not allow create operation
+      await expect(
+        async () => await user1TodoHelper.create(createResultSetName, { ...todoPrivateFields, groupsContent: 'Groups Content' }),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(createResultSetName, 'Mutation'));
+
+      // Create a record with allowed fields, so we can test the update and delete operations.
+      const user1CreateAllowedSet = `
+        ${privateResultSet}
+        ownerContent
+        adminContent
+        groupContent
+      `;
+      const createResult1 = await user1TodoHelper.create(
+        createResultSetName,
+        { ...todoPrivateFields, ownerContent: 'Owner Content', adminContent: 'Admin Content', groupContent: 'Group Content' },
+        user1CreateAllowedSet,
+      );
+      expect(createResult1.data[createResultSetName].id).toBeDefined();
+      expect(createResult1.data[createResultSetName].author).toEqual(userName1);
```

Comment:
nit: the `expect.objectContaining(todoPrivateFields)` can be used instead of comparing each field